src/nvim/misc1.c

```diff
@@ -1292,11 +1293,12 @@ int plines_win_nofold(win_T *wp, linenr_T lnum)
   width = wp->w_width - win_col_off(wp);
   if (width <= 0)
     return 32000;
-  if (col <= width)
+  if (col <= (unsigned int)width)
     return 1;
   col -= width;
   width += win_col_off2(wp);
-  return (col + (width - 1)) / width + 1;
+  assert(col <= INT_MAX);
```

Comment:
Here I was just "mechanically" asserting following cast is safe.
I'm not sure if this is the place to also check for variable "conceptual" limits. We're not changing a persistent variable holding a column number, but just making a calculation of a different thing out of it.
Also, we're adding at most 1, and subtracting at least 1. So, if value returned by by win_linetabsize() is in bounds, so will be the newly calculated value.
I don't have any problem to add the assert if you prefer, though.